frontend/src/pages/user/admin/allReports/[adminReportID].tsx

```diff
@@ -0,0 +1,88 @@
+import '@style/global.css';
+
+import { Col, Row } from 'react-bootstrap';
+import { useParams } from 'react-router-dom';
+
+import SellerItem from '@components/SellerItem';
+import NotFound from '@components/NotFound';
+
+import adminReportData from '@pages/user/admin/adminReportData.json';
+
+interface Sellers {
+  id: number;
+  name: string;
+  imgUrl: string;
+  amount: number;
+}
+
+interface Props {
+  year: number;
+  month: number;
+  sellers: Sellers[];
+  totalAmount: number;
+  id: number;
+}
+
+const AdminReportEach = () => {
+  const reportPageStyle = {
+    borderRadius: '50px 50px 0px 0px',
+    border: '1px solid var(--button_border)',
+    background: 'var(--bg)',
+    boxShadow: '0px 4px 30px 2px var(--title)',
+    padding: ' 7% 12% 10% 12%',
+  };
+
+  const { year, month } = useParams();
+  let data: Props | undefined;
+  adminReportData.findIndex((item) => {
+    if (item.year.toString() === year && item.month.toString() === month) {
+      data = item;
+    }
+  });
```

Comment:
what is this trying to achieve?

Reply:
try to find the corresponding data according to year and month, and assign to `data` variable.
i do so rather than put the return() inside the `if (item.year.toString() === year && item.month.toString() === month)` and map with item is because when change to using api, i can just remove the part of `findIndex`

Reply:
but shouldn't `findIndex()` take in something like `(i: Item) => bool`? I have no idea why typescript accepts this, if you want to mutate the values directly without returning values, you probably should use `forEach()` instead

Reply:
do you mean that if i want to use findIndex, i should get the index first then assign value with the index of that data

Reply:
if I understand what you're trying to do correctly, it should probably be like this:
```js
let data = adminReportData.find((item) => item.year.toString() === year && item.month.toString() === month);
```

Reply:
you're correct i should just use find(). my bad

Reply:
wait i remember why i wrote this
because typescript will tell you change the `let` to `const` if the data will not changed again.
then the const variable needs type so i need to initialize it, just write  with :Props will not work, i though i can't write in one line but now it seems it's work.

Reply:
isn't that because `.find()` has a chance to not find anything even though it isn't impossible in this case? if so you could try type coercing it or provide it with a default value